src/pudl/extract/ferc1.py

```diff
@@ -829,52 +837,69 @@ def extract_xbrl(
         # Attempt to extract both duration and instant tables
         xbrl_table = TABLE_NAME_MAP_FERC1[pudl_table]["xbrl"]
         ferc1_raw_dfs[pudl_table] = {}
+
+        ferc1_xbrl_engine = sa.create_engine(pudl_settings["ferc1_xbrl_db"])
+
         for period_type in ["duration", "instant"]:
-            ferc1_raw_dfs[pudl_table][period_type] = extract_xbrl_generic(
-                ferc1_engine=sa.create_engine(pudl_settings["ferc1_xbrl_db"]),
-                ferc1_settings=ferc1_settings,
-                table_name=f"{xbrl_table}_{period_type}",
-            )
+            if type(xbrl_table) is list and len(xbrl_table) > 1:
```

Comment:
Did you run into trouble trying to make `extract_xbrl_generic` handle both lists and singletons? To me, it seems like we could turn `extract_xbrl_concat` into the new `_generic` function, make sure everything gets passed in as a list, and rename the old `_generic` function to something like `extract_xbrl_single`...

```diff
@@ -835,31 +835,27 @@ def extract_xbrl(
         )

         # Attempt to extract both duration and instant tables
-        xbrl_table = TABLE_NAME_MAP_FERC1[pudl_table]["xbrl"]
-        ferc1_raw_dfs[pudl_table] = {}
+        xbrl_table_or_tables = TABLE_NAME_MAP_FERC1[pudl_table]["xbrl"]
+        if not isinstance(xbrl_table_or_tables, list):
+            xbrl_tables = [xbrl_table_or_tables]
+        else:
+            xbrl_tables = xbrl_table_or_tables

         ferc1_xbrl_engine = sa.create_engine(pudl_settings["ferc1_xbrl_db"])
-
-        for period_type in ["duration", "instant"]:
-            if type(xbrl_table) is list and len(xbrl_table) > 1:
-                ferc1_raw_dfs[pudl_table][period_type] = extract_xbrl_concat(
-                    table_names=xbrl_table,
-                    period=period_type,
-                    ferc1_engine=ferc1_xbrl_engine,
-                    ferc1_settings=ferc1_settings,
-                )
-            else:
-                ferc1_raw_dfs[pudl_table][period_type] = extract_xbrl_generic(
-                    table_name=xbrl_table,
-                    period=period_type,
-                    ferc1_engine=ferc1_xbrl_engine,
-                    ferc1_settings=ferc1_settings,
-                )
+        ferc1_raw_dfs[pudl_table] = {
+            period_type: extract_xbrl_generic(
+                table_names=xbrl_tables,
+                period=period_type,
+                ferc1_engine=ferc1_xbrl_engine,
+                ferc1_settings=ferc1_settings,
+            )
+            for period_type in ["duration", "instant"]
+        }
```

Reply:
We could also make that "singleton-or-list -> list" transformation in `extract_xbrl_generic` itself if we want to keep the API extra stable.

Reply:
I like this idea. We could probably simplify these functions even more, but I'm not going to put too much effort in right now. Also did this for the `extract_dbf` function.